probe-rs/src/config/registry.rs

```diff
@@ -145,112 +145,109 @@ impl Registry {
-                            }
+        // Try get the corresponding chip.
+        let mut selected_family_and_chip = None;
+        let mut exact_matches = 0;
+        let mut partial_matches = Vec::new();
+        for family in self.families.iter() {
+            for variant in family.variants.iter() {
+                if match_name_prefix(&variant.name, name) {
+                    if variant.name.len() == name.len() {
+                        tracing::debug!("Exact match for chip name: {}", variant.name);
+                        exact_matches += 1;
+                    } else {
+                        tracing::debug!("Partial match for chip name: {}", variant.name);
+                        partial_matches.push(variant.name.as_str());
+                        // Only select partial match if we don't have an exact match yet
+                        if exact_matches > 0 {
+                            continue;
                         }
-                        selected_family_and_chip = Some((family, variant));
                     }
+                    selected_family_and_chip = Some((family, variant));
                 }
             }
-            if exact_matches == 0 && partial_matches.len() > 1 {
-                tracing::warn!(
-                    "Ignoring ambiguous matches for specified chip name {}",
-                    name,
-                );
-                let mut suggestions;
-                if partial_matches.len() <= 100 {
-                    suggestions = partial_matches.join(", ");
-                } else {
-                    // prevent too much text being printed if too many matches
-                    suggestions = partial_matches[0..100].join(", ");
-                    suggestions.push_str(&format!(" and {} more", partial_matches.len() - 100))
+        }
+
+        let Some((family, chip)) = selected_family_and_chip else {
+            return Err(RegistryError::ChipNotFound(name.to_string()));
+        };
+
+        if exact_matches == 0 {
+            match partial_matches.len() {
+                0 => {}
+                1 => {
+                    tracing::warn!(
+                        "Found chip {} which matches given partial name {}. Consider specifying its full name.",
+                        chip.name,
+                        name,
+                    );
+                }
+                matches => {
+                    const MAX_PRINTED_MATCHES: usize = 100;
+                    tracing::warn!(
+                        "Ignoring {matches} ambiguous matches for specified chip name {name}"
+                    );
+
+                    let (print, overflow) =
+                        partial_matches.split_at(MAX_PRINTED_MATCHES.min(matches));
+
+                    let mut suggestions = print.join(", ");
+
+                    match overflow.len() {
+                        0 => {}
+                        1 => suggestions.push_str(&format!(", {}", overflow[0])),
```

Comment:
Why do we have this branch? Can't we just increase the MAX_PRINTED_MATCHES by one?

Reply:
I absolutely hate lists that can end with `, and 1 more`, so this just adds a bit of flexibility to the max to avoid it. The amount of times I was actually fed up by "just print the one damn thing, it would fit" is waay too much :)

I.e. if `MAX_PRINTED_MATCHES = 3`:

```
1
1, 2
1, 2, 3
1, 2, 3, 4
1, 2, 3, and 2 more
```

(Also, 100 is probably waay to much to be useful here?)

Reply:
Ah, I see! Okay, fair enough! Maybe a comment would be nice? :)